osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs

```diff
@@ -445,8 +445,13 @@ protected override IEnumerable<MenuItem> GetContextMenuItemsForSelection(IEnumer
             foreach (var item in base.GetContextMenuItemsForSelection(selection))
                 yield return item;
 
-            if (selectedMergeableObjects.Length > 1)
+            if (canMerge(selectedMergeableObjects))
                 yield return new OsuMenuItem("Merge selection", MenuItemType.Destructive, mergeSelection);
         }
+
+        private bool canMerge(IReadOnlyList<OsuHitObject> objects) =>
+            objects.Count > 1
+            && (objects.Any(h => h is Slider)
+                || Precision.DefinitelyBigger(Vector2.DistanceSquared(objects[0].Position, objects[1].Position), 1));
```

Comment:
I disagree. In this case we are trying to make sure that the merged path will have more than 1 length, so I wanted to check that there exist two consecutive circles which are more than 1 distance away from eachother. Checking every consecutive pair seemed like too much work so i just check the first pair. If you change this to the first and last object instead then its no longer consecutive and it will have a different behaviour. For instance a merge with 3 circles going back and forth with the first and last circles overlapping would be blocked by this.

Reply:
While I saw what the intent behind only checking first and second was as stated above, and so I didn't comment on it initially, only checking first and last would prevent cases like this technically "legal" merge:

![image](https://user-images.githubusercontent.com/20418176/187285453-98fd7f69-9828-4987-920d-46879be8abc6.png)

So I'd say that if we care about the accuracy of this check to that degree wherein the above should be allowed, then every pair needs to be examined.

Reply:
okay i can do that with this monstrosity of a linq expression `objects.Zip(objects.Skip(1), (h1, h2) => Precision.DefinitelyBigger(Vector2.DistanceSquared(h1.Position, h2.Position), 1)).Any(x => x)`

Reply:
Works for me.